src/evict/README.md

```diff
@@ -0,0 +1,49 @@
+# Eviction
+
+### Overview
+
+Eviction process ensures that the cache in WiredTiger is efficiently managed and it stays within user-defined boundaries. These boundaries are set through **target** and **trigger thresholds** for total, dirty, and update content. When the cache exceeds these limits, eviction processes begins to reduce the cache content.
+
+### Key Parameters
+
+WiredTiger offers multiple configuration options to manage the eviction of pages from the cache. These settings can be adjusted during database opening (`wiredtiger_open`) or reconfigured later (`WT_CONNECTION::reconfigure`).
+
+| Parameter               | Description                                                                                              |
+| ----------------------- | -------------------------------------------------------------------------------------------------------- |
+| **`eviction_target`**    | The target percentage of cache usage that eviction tries to maintain.                                  |
+| **`eviction_trigger`**   | When cache usage exceeds this threshold, **application threads**  start assisting in eviction.     |
+| **`eviction_dirty_target`** | The target percentage of **dirty** cache usage that eviction tries to maintain.                      |
+| **`eviction_dirty_trigger`** | When cache usage exceeds this **dirty** threshold, **application threads**  start assisting in eviction.   |
+| **`eviction_updates_target`** | The target percentage of **updates** cache usage that eviction tries to maintain.                 |
+| **`eviction_updates_trigger`** | When cache usage exceeds this **updates** threshold, **application threads** start assisting in eviction.                   |
+
+> **Note:** Target sizes must always be lower than their corresponding trigger sizes.
+
+
+### Eviction Process
+
+The eviction process involves three components:
+
+- **Eviction Server**: The primary background process that commences when the **target thresholds** are reached. It identifies evictable candidates, places them in eviction queues and sorts them based on the **Least Recently Used (LRU)** algorithm.
+- **Eviction Worker Threads**: These threads pop pages from the queues and evicts them. They run in the background to assist the server.
```

Comment:
This should mention a little more about the behaviour, ie what does a thread do? (reconcile, write to disk and evict from memory)

Should this also say something about parallelism constraints (ie the number of threads is limited by X, or should be balanced by Y or something like that)

Reply:
Thanks, adding info about parallelism constraints sounds good. Updated in [ed8de00](https://github.com/wiredtiger/wiredtiger/pull/11096/commits/ed8de00ccecdb9267c5f4b92b2626b57f9a2ccbd).

However, in terms of there behaviour, these threads pop pages from the queue and evict them. As part of eviction reconciliation may be called. I left out any mention of reconciliation from this doc for simplicity sake.